src/main/java/com/mongodb/jdbc/MongoSQLResultSet.java

```diff
@@ -132,136 +114,65 @@ protected Object getObject(BsonValue o, int columnType) throws SQLException {
             return null;
         }
         switch (columnType) {
-            case Types.ARRAY:
-                // not supported
-                break;
             case Types.BIGINT:
```

Comment:
BIGINT should return a Long. Otherwise we might encounter overflow issues. Especially since we correctly map BSON_LONG to BIGINT.

Reply:
done